includes/transformer/class-base.php

```diff
@@ -46,9 +61,9 @@ public function __construct( $object ) {
 	 * @return Activitypub\Activity\Base_Object
 	 */
 	public function to_object() {
-		$object = new Base_Object();
+		$this->activitypub_object = new Base_Object();
 
-		$vars = $object->get_object_var_keys();
+		$vars = $this->activitypub_object->get_object_var_keys();
```

Comment:
why storing it in a local var?

Reply:
it is not used outside of this function, so it might be more confusing than it would help?!?

Reply:
You are right. From this Pull Requests point of view custom transformers might override the to_object function anyway. And they could also define their own protected property for storing the already processed part of the object. 

I would suggest reverting this part of the code and deal with the performance optimizations by reusing already transformed parts later. One would also then have to take care of the race-conditions which properties gets transformed when.